File: 2085-count-common-words-with-one-occurrence/2085-count-common-words-with-one-occurrence.js
/**
 * @param {string[]} words1
 * @param {string[]} words2
 * @return {number}
 */
var countWords = function(words1, words2) {
    const map = {}
    for (const w of words1) {
        map[w] = map[w] + 1 || 1
    }
    for (const w of words2) {
        if (w in map && map[w] < 2) {
            map[w]--
        }
    }
    let res = 0
    for (const key in map) {
        if (map[key] === 0) {
            res++
        }
    }
    return res
};

File: 2085-count-common-words-with-one-occurrence/2085-count-common-words-with-one-occurrence.py
class Solution:
    def countWords(self, words1: List[str], words2: List[str]) -> int:
        mp, res = {}, 0
        for w in words1:
            mp[w] = mp.get(w, 0) + 1
        for w in words2:
            if w in mp and mp[w] < 2:
                mp[w] -= 1
        for key in mp:
            if mp[key] == 0:
                res += 1
        return res
            